static/slider.js

```diff
@@ -143,4 +180,51 @@
     function dateToTime(date) {
         return new Date(date).getTime() / 1000;
     }
+
+    /**
+     * Change heat and star td element for single cell.
+     * @param {Integer} size - Quantity of observations.
+     * @param {String} color - New color for sublevel.
+     * @param {String} starsClass - Class to find star element.
+     * @param {String} heatClass - Class to find heat element.
+     * @param {Integer} construct - ID of current construct.
+     */
+    function changeElems(size, color, starsClass, heatClass, construct) {
+        let elem = $(`.star-chart-4-table-${construct}`)
+            .find(heatClass)
+            .attr('bgcolor', color);
+        elem[0].dataset.color = color;
+
+        elem = $(`.star-chart-4-table-${construct}`)
+            .find(starsClass)
+            .html(size);
+        elem[0].dataset.stars = size;
+    }
+
+    /**
+     * Change values for cells in verticalStarChart map.
+     * @param {Object} observations 
+     * @param {Object} allStars - All observations per construct.
+     */
+    function swapVerticalStarChart(observations, allStars) {
+        for (var construct in observations) {
+            if (window.mergedConstructs.includes(construct)) {
+                Object.keys(observations[construct]).forEach((course, index) => {
+                    Object.keys(observations[construct][course]).forEach((sublevel, j) => {
+                        const size = observations[construct][course][sublevel].length;
```

Comment:
It's 5 levels of indentation. IMO you could add function which handles whole stuff from inside of `if`.